impor library-library yg diperlukan

In [3]:
import pandas as pd
import numpy as np

pandas mempunyai 2 objec yaitu series dan data frame

Object Series

Object series mempunyai satu dimensi data.. tidak mempunyai nama column karena hanya memiliki satu column saja.. Dan mempunyai Indeks

In [17]:
data = [0.25,0.5,0.75,1]

In [18]:
print(data)

[0.25, 0.5, 0.75, 1]


merubah data kedalam bentuk series

In [19]:
data = pd.Series(data)

In [20]:
data

0    0.25
1    0.50
2    0.75
3    1.00
dtype: float64

Task 1 Bikin sebuah list, ubah ke dalam bentuk series

In [25]:
data1 = ["Novan","Ardi","Mala","Renata"]
print(data1)

['Novan', 'Ardi', 'Mala', 'Renata']


In [26]:
data1 = pd.Series(data1)
data1

0     Novan
1      Ardi
2      Mala
3    Renata
dtype: object

convert data series menjadi array

In [21]:
data.values

array([0.25, 0.5 , 0.75, 1.  ])

menampilkan indeks 

indeks berupa range, dimana titik start nya itu inklusif dalam range dan titik stopnya itu eksklusif dari range

In [22]:
data.index

RangeIndex(start=0, stop=4, step=1)

In [23]:
list(range(1,10))

[1, 2, 3, 4, 5, 6, 7, 8, 9]

cara memanggil data

In [24]:
data[2]

0.75

Task 2 convert series ke bentuk array, tampilkan indexnya, slicing

In [27]:
data1.values

array(['Novan', 'Ardi', 'Mala', 'Renata'], dtype=object)

In [28]:
data1.index

RangeIndex(start=0, stop=4, step=1)

In [29]:
data1[0]

'Novan'

---ada tulisannya---

In [30]:
data = pd.Series([0.25,0.5,0.75,1], index=['a','b','c','d'])

In [31]:
data

a    0.25
b    0.50
c    0.75
d    1.00
dtype: float64

In [32]:
data.index

Index(['a', 'b', 'c', 'd'], dtype='object')

panggil datanya

In [33]:
data['a']

0.25

ini adalah data selection 

walaupun kita sudah mempbuat indeks eksplisit, kita masih bisa memanggil indeks implisitnya

In [34]:
data[3]

1.0

task 3 bikin indeks eksplisit, tampilkan indeks, panggil indeks eksplisit, panggil indeks implisit

In [35]:
data1 = pd.Series(["Novan", "Ardi", "Mala", "Renata"], index=['a','b','c','d'])
data1

a     Novan
b      Ardi
c      Mala
d    Renata
dtype: object

In [38]:
data1.index

Index(['a', 'b', 'c', 'd'], dtype='object')

In [39]:
data1['a']

'Novan'

In [40]:
data1[0]

'Novan'

ketika indeks eksplisit dan indeks implisitnya ada yg sama, maka dia hanya akan mengandalkan indeks eksplisitnya.

In [45]:
data_2 = pd.Series([0.25,0.5,0.75,1], index=[4,5,6,7])

In [47]:
data_2[4]

0.25

In [52]:
data_2[5]

0.5

In [50]:
data_2

4    0.25
5    0.50
6    0.75
7    1.00
dtype: float64

Task 4 bikin indeks eksplisit yang ada kesamaan dgn indeks implisit, bikin series dari list yg isinya 10 elemen

In [55]:
data3 = pd.Series([1,3,5,7,9,11,13,15,17,19], index=[4,2,3,5,6,1,11,10,13,16])
data3

4      1
2      3
3      5
5      7
6      9
1     11
11    13
10    15
13    17
16    19
dtype: int64

In [56]:
data3[11]

13

In [57]:
data3[0]

KeyError: 0

kita akan coba melakukan data slicing

In [59]:
data = pd.Series([0.25,0.5,0.75,1], index=['a','b','c','d'])
data

a    0.25
b    0.50
c    0.75
d    1.00
dtype: float64

misalnya kita akan memanggil dari data b sampai data c

In [61]:
data['b':'c']

b    0.50
c    0.75
dtype: float64

tapi bila kita slicing indeks implisitnya, maka hanya akan muncul titik start nya.. karena indeks implisit berupa range

In [62]:
data[1:2]

b    0.5
dtype: float64

Task 5 bikin series minimal 10 elemen, bikin indeks eksplisit, slicing eksplisit, slicing implisit pake 2 param dan 3 param

In [65]:
data4 = pd.Series([1,3,5,7,9,11,13,15,17,19], index=['a','b','c','d','e','f','g','h','i','j'])

In [64]:
data4['b':'i']

b     3
c     5
d     7
e     9
f    11
g    13
h    15
i    17
dtype: int64

In [66]:
data4[2:9]

c     5
d     7
e     9
f    11
g    13
h    15
i    17
dtype: int64

In [67]:
data4[2:9:3]

c     5
f    11
i    17
dtype: int64

loc dan iloc



In [4]:
data_2 = pd.Series([0.25,0.5,0.75,1], index=[2,5,3,7])
data_2

2    0.25
5    0.50
3    0.75
7    1.00
dtype: float64

ketika indeks eksplisit dan implisit ada yang sama maka akan terjadi inkonsistensi pada case diatas.

untuk mengatasi inkonsistensi tersebut maka kita akan gunakan kaidah loc dan iloc..
loc adalah untuk memanggil indeks eksplisitnya..
iloc adalah untuk memanggil indeks implisitnya..

In [6]:
data_2.loc[3] #selecting inndex explisit

0.75

In [5]:
data_2.loc[2:3] #slicing index explisit

2    0.25
5    0.50
3    0.75
dtype: float64

In [6]:
data_2.iloc[3] #selecting index implisit

1.0

In [7]:
data_2.iloc[2:3] #slicing index implisit

3    0.75
dtype: float64

Task 1 bikin series, bikin index eksplisitnya ada yg sama dengan indeks python, panggil loc, panggil iloc

In [8]:
data3 = pd.Series([1,3,5,7,9,11,13,15,17,19], index=[4,2,3,5,6,1,11,10,13,16])
data3

4      1
2      3
3      5
5      7
6      9
1     11
11    13
10    15
13    17
16    19
dtype: int64

In [9]:
data3.loc[5]

7

In [10]:
data3.iloc[5]

11

bikin list menggunakan dictionary

In [11]:
dict_populasi = {'Jakarta':750,
                 'Bogor':490,
                 'Depok':350,
                 'Tangerang':270,
                 'Bekasi':670}

In [12]:
dict_populasi

{'Jakarta': 750, 'Bogor': 490, 'Depok': 350, 'Tangerang': 270, 'Bekasi': 670}

In [14]:
#transformasi dictionary ke series
populasi = pd.Series(dict_populasi)
populasi

Jakarta      750
Bogor        490
Depok        350
Tangerang    270
Bekasi       670
dtype: int64

In [15]:
populasi.loc['Depok']

350

In [16]:
populasi.iloc[2]

350

Task 2

In [17]:
tinggi_badan = {'Ronaldo':185,
               'Messi':179,
               'Neymar':182,
               'Mbappe':180,
               'Haaland':189}
tinggi_badan

{'Ronaldo': 185, 'Messi': 179, 'Neymar': 182, 'Mbappe': 180, 'Haaland': 189}

In [19]:
populasi2 = pd.Series(tinggi_badan)
populasi2

Ronaldo    185
Messi      179
Neymar     182
Mbappe     180
Haaland    189
dtype: int64

In [20]:
populasi2.loc['Haaland']

189

In [21]:
populasi2.iloc[4]

189

---

In [25]:
dict_luas = {'Jakarta':737,
            'Bogor':325,
            'Depok':247,
            'Tangerang':302,
            'Bekasi':355}
luas = pd.Series(dict_luas)
luas

Jakarta      737
Bogor        325
Depok        247
Tangerang    302
Bekasi       355
dtype: int64

# Data Frame

Data Frame adalah kumpulan data series, dengan minimal satu buah series

In [45]:
daerah = pd.DataFrame({'populasi':populasi, 'luas daerah':luas})
daerah

populasi  luas daerah
Jakarta         750          737
Bogor           490          325
Depok           350          247
Tangerang       270          302
Bekasi          670          355

In [27]:
daerah['luas daerah']

Jakarta      737
Bogor        325
Depok        247
Tangerang    302
Bekasi       355
Name: luas daerah, dtype: int64

In [28]:
daerah['luas daerah']['Jakarta']

737

In [46]:
daerah['populasi'].iloc[0:3]

Jakarta    750
Bogor      490
Depok      350
Name: populasi, dtype: int64

Task 3

In [41]:
tinggi_badan = {'Ronaldo':185,
               'Messi':179,
               'Neymar':182,
               'Mbappe':180,
               'Haaland':189}
tb = pd.Series(tinggi_badan)

In [40]:
berat_badan = {'Ronaldo':72,
               'Messi':66,
               'Neymar':68,
               'Mbappe':65,
               'Haaland':75}
bb = pd.Series(berat_badan)

In [39]:
dict_goal = {'Ronaldo':24,
               'Messi':23,
               'Neymar':20,
               'Mbappe':27,
               'Haaland':29}
goal = pd.Series(dict_goal)

In [47]:
pemain = pd.DataFrame({'tinggi pemain':tb, 'berat pemain':bb,"goal":goal})
pemain

tinggi pemain  berat pemain  goal
Ronaldo            185            72    24
Messi              179            66    23
Neymar             182            68    20
Mbappe             180            65    27
Haaland            189            75    29

In [49]:
pemain['goal'].iloc[2:5]

Neymar     20
Mbappe     27
Haaland    29
Name: goal, dtype: int64

In [38]:
pemain['tinggi badan']

Ronaldo    185
Messi      179
Neymar     182
Mbappe     180
Haaland    189
Name: tinggi badan, dtype: int64

In [4]:
pemain['goal musim ini']['Mbappe']

NameError: name 'pemain' is not defined

In [6]:
#kita load data titanic

df = pd.read_csv('Titanic.csv')

In [7]:
#melihat dari data teratas

df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [8]:
#melihat info data

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Task 1 tampilkan 10 data teratas dan terbawah, kolom mana saja yg berisi missing values, kolom mana saja yg tipe datanya belum sesuai

In [11]:
df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [12]:
df.tail(10)

PassengerId  Survived  Pclass                                      Name  \
881          882         0       3                        Markun, Mr. Johann   
882          883         0       3              Dahlberg, Miss. Gerda Ulrika   
883          884         0       2             Banfield, Mr. Frederick James   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
881    male  33.0      0      0            349257   7.8958   NaN        S  
882  female  22.0      0      0              7552  10.5167   NaN        S  
883    male  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884    male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
885  female  39.0      0      5            382652  29.1250   NaN        Q  
886    male  27.0      0      0            211536  13.0000   NaN        S  
887  female  19.0      0      0            112053  30.0000   B42        S  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  
889    male  26.0      0      0            111369  30.0000  C148        C  
890    male  32.0      0      0            370376   7.7500   NaN        Q

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


kolom yg berisi missing value : age, cabin, embarked, karena di info, non null nya kurang dari 891

kolom yg data tipe nya belum sesuai : age, harusnya pakai int

In [15]:
#melihat jumlah baris dan jumlah column

df.shape

(891, 12)

In [17]:
#melihat column

df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [18]:
#melihat index

df.index

RangeIndex(start=0, stop=891, step=1)

In [19]:
#menampilkan informasi dari column-column yg berbentuk angka

df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

Task 2 gali insight (selain kolom age)

##### Insight Kolom Fare
* Harga tiket paling mahal adalah 512.3292
* Harga tiket paling murah adalah 0 (asumsi mata uangnya adalah dollar)
* Rata-rata harga tiketnya adalah 32.204208
* Semua penumpang kapal membeli tiket karena jumlah count baris sama dengan range index

##### Insight Kolom Survived
* Rata-rata kemungkinan hidup penumpang titanic adalah 0,38 atau 38%
* Semua orang dapat diidentifikasi status keselamatannya (masih hidup atau sudah meninggal)